### PR TITLE
mgr/dashboard: Angular v11 Migration

### PR DESCRIPTION
**Pull request by nizamial09**
Changelog: https://v11.angular.io/guide/updating-to-version-11#changes-and-deprecations-in-version-11

1. group is deprecated: This API is not typesafe and can result in issues with Closure Compiler renaming. Use the `FormBuilder#group` overload with `AbstractControlOptions` instead.
2. cypress/integration/rgw/users.po.ts(113,23): error TS2345: Argument of type 'void' is not assignable to parameter of type 'string'.
3. Cannot find module 'isomorphic-form-data' from 'node_modules/swagger-ui/dist/swagger-ui.js' -> had to introduce `'isomorphic-form-data'` dependency to fix this.
4. Changed `xi18n` to `extract-i18n` since it is deprecated from 11. see -> https://blog.ninja-squad.com/2020/11/11/angular-cli-11.0/
5. Remove i18n Warning about duplicate id
6. Directives in the @angular/forms package used to have any[] as a type of validators and asyncValidators arguments in constructors. Now these arguments are properly typed

Fixes: https://tracker.ceph.com/issues/50268
Signed-off-by: Nizamudeen A <nia@redhat.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
jenkins test make check

**Comment by aaryanporwal** [comment]
jenkins test dashboard

**Review by avanthakkar — APPROVED** [review]
LGTM! 
Just 1 thing though, in case u have seen it:
- `Directives in the @angular/forms package used to have any[] as a type of validators and asyncValidators arguments in constructors. Now these arguments are properly typed, so if your code relies on directive constructor types it may require some updates to improve type safety`
^^ This maybe related to some part of our code, but I found just at 1 place where maybe we need to change 
https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-setting/iscsi-setting.component.ts#L22
it's a very minor thing..maybe even doesn't affect, but just mentioning.

**Comment by nizamial09** [comment]
> LGTM! Just 1 thing though, in case u have seen it:
> 
> * `Directives in the @angular/forms package used to have any[] as a type of validators and asyncValidators arguments in constructors. Now these arguments are properly typed, so if your code relies on directive constructor types it may require some updates to improve type safety`
>   ^^ This maybe related to some part of our code, but I found just at 1 place where maybe we need to change
>   https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-setting/iscsi-setting.component.ts#L22
>   it's a very minor thing..maybe even doesn't affect, but just mentioning.

Thanks @avanthakkar Updated the PR.

**Review by pereman2 — APPROVED** [review]

File: src/pybind/mgr/dashboard/frontend/package.json [hunk 1]
```diff
@@ -104,27 +104,28 @@
...
     "@applitools/eyes-cypress": "^3.22.0",
     "@compodoc/compodoc": "1.1.11",
     "@types/jest": "26.0.14",
     "@types/lodash": "4.14.161",
     "@types/node": "12.12.62",
     "@types/simplebar": "5.1.1",
-    "@types/swagger-ui": "3.47.0",
-    "codelyzer": "6.0.1",
+    "@types/swagger-ui": "3.52.0",
+    "codelyzer": "^6.0.0",
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/frontend/package.json:122]: @nizamial09 I know it's too late but we use pinned versions for stability purposes.

Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/package.json:122]: @alfonsomthd Oh okay, I think its the angular migration script's doing.  But I can change it here maybe https://github.com/ceph/ceph/pull/43404 (just something I started on weekend ;) )

**Pull request merged**